plasmapy/particles/tests/test_exceptions.py

```diff
@@ -0,0 +1,1118 @@
+        ParticleError,
+    ),
+    (
+        "ion",
+        IonizationStateCollection,
+        [],
+        {"inputs": {"H": [0.1, 0.9], "He+": [0.0, 0.9, 0.1]}},
+        ParticleError,
+    ),
+    (
+        "repeat elements",
+        IonizationStateCollection,
+        [],
+        {"inputs": {"H": [0.1, 0.9], "hydrogen": [0.2, 0.8]}},
+        ParticleError,
+    ),
+    (
+        "isotope of element",
+        IonizationStateCollection,
+        [],
+        {"inputs": {"H": [0.1, 0.9], "D": [0.2, 0.8]}},
+        ParticleError,
+    ),
+    (
+        "negative abundance",
+        IonizationStateCollection,
+        [],
+        {
+            "inputs": {"H": [0.1, 0.9], "He": [0.4, 0.5, 0.1]},
+            "abundances": {"H": 1, "He": -0.1},
+        },
+        ParticleError,
+    ),
+    (
+        "imaginary abundance",
+        IonizationStateCollection,
+        [],
+        {
+            "inputs": {"H": [0.1, 0.9], "He": [0.4, 0.5, 0.1]},
+            "abundances": {"H": 1, "He": 0.1j},
+        },
+        ParticleError,
+    ),
+    (
+        "wrong density units",
+        IonizationStateCollection,
+        [],
+        {
+            "inputs": {"H": [10, 90] * u.m ** -3, "He": [0.1, 0.9, 0] * u.m ** -2},
+            "abundances": {"H": 1, "He": 0.1},
+        },
+        ParticleError,
+    ),
+    (
+        "abundance redundance",
+        IonizationStateCollection,
+        [],
+        {
+            "inputs": {"H": [10, 90] * u.m ** -3, "He": [0.1, 0.9, 0] * u.m ** -3},
+            "abundances": {"H": 1, "He": 0.1},
+        },
+        ParticleError,
+    ),
+    (
+        "abundance contradiction",
+        IonizationStateCollection,
+        [],
+        {
+            "inputs": {"H": [10, 90] * u.m ** -3, "He": [0.1, 0.9, 0] * u.m ** -3},
+            "abundances": {"H": 1, "He": 0.11},
+        },
+        ParticleError,
+    ),
+    (
+        "kappa too small",
+        IonizationStateCollection,
+        [],
+        {"inputs": ["H"], "kappa": 1.499999},
+        ParticleError,
+    ),
+    (
+        "negative n",
+        IonizationStateCollection,
+        [],
+        {"inputs": ["H"], "n0": -1 * u.cm ** -3},
+        ParticleError,
+    ),
+    (
+        "negative T_e",
+        IonizationStateCollection,
+        [],
+        {"inputs": ["H-1"], "T_e": -1 * u.K},
+        ParticleError,
+    ),
+]
+
+tests_for_exceptions = (
+    tests_for_exceptions_IonizationState
+    + tests_for_exceptions_IonizationStateCollection
+)
+
+
+@pytest.mark.parametrize(
+    ["test_name", "tested_object", "args", "kwargs", "expected_exception"],
+    tests_for_exceptions,
+)
+def test_named_tests_for_exceptions(
+    test_name, tested_object, args, kwargs, expected_exception
+):
+    """
+    Test that appropriate exceptions are raised for inappropriate inputs
+    to `IonizationState` or `IonizationStateCollection`
+    """
+    print(test_name)  # TODO find better ways for this
```

Comment:
```suggestion
```
Maybe drop this line?

Reply:
Or perhaps an explicit `pytest.fail(f"problem with {test_name}")` at the end of the `with pytest.raises` block?